extensions/MYTAditya/CommentPlus.js

```diff
@@ -0,0 +1,376 @@
+            text: "[COLOR] // [COMMENT]",
+            arguments: {
+              COMMENT: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "comment",
+              },
+              COLOR: {
+                type: Scratch.ArgumentType.COLOR,
+                defaultValue: "#037b03",
+              },
+            },
+          },
+          {
+            opcode: "commentReporterP7",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "// [COMMENT] [COLOR]",
+            arguments: {
+              COMMENT: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "comment",
+              },
+              COLOR: {
+                type: Scratch.ArgumentType.COLOR,
+                defaultValue: "#037b03",
+              },
+            },
+          },
+          {
+            opcode: "commentBooleanP7",
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: "// [COMMENT] [COLOR]",
+            arguments: {
+              COMMENT: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "comment",
+              },
+              COLOR: {
+                type: Scratch.ArgumentType.COLOR,
+                defaultValue: "#037b03",
+              },
+            },
+          },
+          {
+            opcode: "commentCPlusPlus",
+            blockType: Scratch.BlockType.CONDITIONAL,
+            branchCount: 2,
+            text: "// [COMMENT]",
+            arguments: {
+              COMMENT: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "comment",
+              },
+            },
+          },
+        ],
+      };
+    }
+
+    commentHat() {
+      // no-op
+    }
+
+    commentCap(args) {
+      // no-op
+    }
+
+    commentCommand() {
+      // no-op
+    }
+
+    commentC(args, util) {
+      return true;
+    }
+
+    commentReporter(args) {
+      return args.INPUT;
+    }
+
+    commentBoolean(args) {
+      return args.INPUT || false;
+    }
+
+    commentReporterPPP(args) {
+      return args.INPUT;
+    }
+
+    commentBooleanPPP(args) {
+      return args.INPUT || false;
+    }
+
+    commentReporterPPPP(args) {
+      return args.INPUT || false;
+    }
+
+    commentBooleanPPPP(args) {
+      return args.INPUT || false;
+    }
+
+    commentBooleanPlus(args) {
+      // no-op
+    }
+
+    commentReporterPlus(args) {
+      // no-op
+    }
+
+    commentCap() {
+      // no-op
+    }
+
+    commentCPlus() {
+      // no-op
+    }
+    
+    commentReporterPlusPlus(args) {
+      return args.INPUT || false;
+    }
+
+    commentBooleanPlusPlus(args) {
+      return args.INPUT || false;
+    }
+    
+    commentReporterP6(args) {
+      return args.COLOR;
+    }
+
+    commentBooleanP6(args) {
+      return args.COLOR || false;
+    }
+    
+    commentReporterP7(args) {
+      return args.COLOR;
+    }
+
+    commentBooleanP7(args) {
```

Comment:
I don't understand where or why you'd need a color input inside of a boolean.

Reply:
Yeah you're right because reporters could do that. You can say, I just created it for Boolean shaped block. If you say, I would remove Booleans which had colors strings.

Reply:
In an ideal world you'd just remove **all** of the color blocks